include/GafferImage/CollectImages.h

```diff
@@ -0,0 +1,92 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2017, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef GAFFERIMAGE_REPLICATELAYER_H
+#define GAFFERIMAGE_REPLICATELAYER_H
+
+#include "Gaffer/StringPlug.h"
+
+#include "GafferImage/ImageProcessor.h"
+
+namespace GafferImage
+{
+
+class CollectImages : public ImageProcessor
+{
+
+	public :
+
+		CollectImages( const std::string &name=defaultName<CollectImages>() );
+		virtual ~CollectImages();
+
+		IE_CORE_DECLARERUNTIMETYPEDEXTENSION( GafferImage::CollectImages, CollectImagesTypeId, ImageProcessor );
+
+		Gaffer::StringVectorDataPlug *layerNamesPlug();
+		const Gaffer::StringVectorDataPlug *layerNamesPlug() const;
+
+		Gaffer::StringPlug *layerContextVariablePlug();
```

Comment:
Any objection to just calling this "layerVariable"? That would match the naming we already have in the Loop, ContextVariables and Wedge nodes. Oh, in case it's not clear, I'm talking about the plug being called "layerVariable", the method would still be called `layerVariablePlug()`.